### ISSUE TITLE
Pub/Sub Publisher requires additional auth dependency on App Engine Standard J8

### ISSUE THREAD

**Issue by jabubake** [GoogleCloudPlatform/google-cloud-java#2144]
Publisher times out on App Engine Standard Java 8, work around is to include 
```
<dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>com.google.auth</groupId>
        <artifactId>google-auth-library-credentials</artifactId>
        <version>0.7.0</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
```

**Comment by ludoch** [comment]
Do we fully understand now (i.e we are not going via the appIdentify GAE API path anymore but a new path using the missing google-auth-library-credentials classes?)
Why don't we have a user friendly error message for this case? Are we swallowing exceptions?

**Comment by lesv** [comment]
The customer wrote me and said:

Did an experiment bypassing the App Default Credentials on AppEngine:

See https://github.com/hienle-hps/getting-started-java/blob/pubsub_manual_token/appengine-standard-java8/helloworld/src/main/java/com/example/gaejava8standard/HelloAppEngine.java#L70

Steps:
1. Use `gcloud auth application-default login` on local box and run Main.java (see above project) to dump your local bearer token.
3. Replace `TOKEN` in the code above with this token and then `gradle appengineDeploy`.

Results:
The google-cloud PubSub client works.

**Comment by jabubake** [comment]
Fixed via #2284
